src/Symfony/Component/Console/Formatter/OutputFormatter.php

```diff
@@ -263,8 +263,12 @@ private function applyCurrentStyle(string $text, string $current, int $width, in
         }
 
         $lines = explode("\n", $text);
-        if ($width === $currentLineLength = \strlen(end($lines))) {
-            $currentLineLength = 0;
+
+        foreach ($lines as $line) {
+            $currentLineLength += \strlen($line);
+            if ($width <= $currentLineLength) {
+                $currentLineLength = 0;
+            }
```

Comment:
Can you explain the new logic? I have a hard time figuring it out.

Reply:
The problems occurred when when you use style inside the line wtih `colspan` and `maxSize`:
On the method `applyCurrentStyle` the count of char was broken. 
Principaly the method explode the words, applyStyle and implode. Sometime you can more than one style inside the line or break a word with style,  but the old code count only the `end($lines)` and that's why the line was not brake at the good length. 
now i count every line, empty or not, and i change the `$currentLineLength` at the good lenght.

Reply:
Can you add a test for OutputFormatter (independently of the one for table)?

Reply:
Test Added
https://github.com/symfony/symfony/commit/1cf9659b5f14d2cd3167c9f8bfba6b440c247cfb#diff-fabb4589bcf7453c1158e6a61a9c39f3R335

Reply:
This PR could not be rebased on 3.4. Because the issues coming when we use the method `Symfony\Component\Console\Helper\Table::setColumnMaxWidth line:231` 
this method was added only on 4.2 https://github.com/symfony/symfony/pull/28373